sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/download/http/HttpDownloadFileTask.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+package com.microsoft.appcenter.distribute.download.http;
+
+import android.net.TrafficStats;
+import android.net.Uri;
+import android.os.AsyncTask;
+
+import com.microsoft.appcenter.http.TLS1_2SocketFactory;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.lang.ref.WeakReference;
+import java.net.URL;
+import java.net.URLConnection;
+
+import javax.net.ssl.HttpsURLConnection;
+
+import static com.microsoft.appcenter.http.HttpUtils.CONNECT_TIMEOUT;
+import static com.microsoft.appcenter.http.HttpUtils.READ_TIMEOUT;
+import static com.microsoft.appcenter.http.HttpUtils.THREAD_STATS_TAG;
+import static com.microsoft.appcenter.http.HttpUtils.WRITE_BUFFER_SIZE;
+
+/**
+ * <h3>Description</h3>
+ * <p>
+ * Internal helper class. Downloads an .apk from AppCenter and stores
+ * it on external storage. If the download was successful, the file
+ * is then opened to trigger the installation.
+ **/
+class HttpDownloadFileTask extends AsyncTask<Void, Void, Void> {
+
+    /**
+     * Maximal number of allowed redirects.
+     */
+    private static final int MAX_REDIRECTS = 6;
+
+    private static final int UPDATE_PROGRESS_BYTES_COUNT = 128 * 1024;
+
+    private final WeakReference<HttpConnectionReleaseDownloader> mDownloader;
+
+    private Uri mDownloadUri;
+
+    /**
+     * Path to the downloading apk.
+     */
+    private File mTargetFile;
+
+    HttpDownloadFileTask(HttpConnectionReleaseDownloader downloader, Uri downloadUri, File targetFile) {
+        mDownloader = new WeakReference<>(downloader);
+        mDownloadUri = downloadUri;
+        mTargetFile = targetFile;
+    }
+
+    @Override
+    protected Void doInBackground(Void... args) {
+        try {
+            File directory = mTargetFile.getParentFile();
+            if (directory == null || !(directory.exists() || directory.mkdirs())) {
+                throw new IOException("Could not create the directory for file:" + mTargetFile.getAbsolutePath());
+            }
+            if (mTargetFile.exists()) {
+
+                //noinspection ResultOfMethodCallIgnored
+                mTargetFile.delete();
+            }
+
+            /* Create connection */
+            URL url = new URL(mDownloadUri.toString());
+            TrafficStats.setThreadStatsTag(THREAD_STATS_TAG);
+            URLConnection connection = createConnection(url, MAX_REDIRECTS);
+            connection.connect();
+            String contentType = connection.getContentType();
+            if (contentType != null && contentType.contains("text")) {
```

Comment:
Why testing for only text? Does not seem very useful.

Reply:
It's the logic from HockeyApp

Reply:
That's not a good enough reason though :D